translate get_fuchsia_source.md file (#165)
FuchsiaOS/FuchsiaOS-docs-zh_CN: get-started/get_fuchsia_source.md

<!--
# Download the Fuchsia source code
-->
# 下载 Fuchsia 源码
<!--
This guide provides instructions on how to download the
Fuchsia source code and set up the Fuchsia development
environment on your machine.
-->

> 这篇文章将指导您下载 Fuchsia 源码以及在您的机器上部署开发环境。

Reviewer: ```suggestion
本指南说明了如何下载 Fuchsia 源码以及在您的机器上部署开发环境。
```

<!--
The steps are:
-->
步骤如下：
<!--
1. [Perform a preflight check](#perform-a-preflight-check).
2. [Install prerequisite packages](#install-prerequisite-packages).
3. [Download the Fuchsia source code](#download-the-fuchsia-source-code).
4. [Set up environment variables](#set-up-environment-variables).
5. [Configure firewall rules (Optional)](#configure-firewall-rules).
-->

> 1. [前期检查](#perform-a-preflight-check).

Reviewer: 同下。

> 2. [安装用到的软件包](#install-prerequisite-packages).

Reviewer: 同下。

3. [下载 Fuchsia 源码](#download-the-fuchsia-source-code).
4. [设置环境变量](#set-up-environment-variables).

> 5. [配置防火墙（可选项）](#configure-firewall-rules).

Reviewer: ```suggestion
5. [配置防火墙规则（可选）](#configure-firewall-rules).
```

<!--
## 1. Perform a preflight check {#perform-a-preflight-check}
-->

> ## 1. 前期检查 {#perform-a-preflight-check}

Reviewer: ```suggestion
## 1. 执行预检查 {#perform-a-preflight-check}
```
感觉还是要把整句翻出来。

<!--
Fuchsia provides a preflight check tool
([`ffx platform preflight`][ffx-platform-preflight])
that examines your machine and informs you of any issues that may
affect building Fuchsia from source on the machine.
-->

> Fuchsia 提供了一个自动检测工具（[`ffx platform preflight`][ffx-platform-preflight]），这个工具可以测试您的机器，并会在发现任何影响您编译 Fuchsia 源码的问题时通知您。

Reviewer: ```suggestion
Fuchsia 提供了一个预检工具（[`ffx platform preflight`][ffx-platform-preflight]），这个工具可以测试您的机器，并会通知您任何在该机器上构建 Fuchsia 源码时会受影响的问题。
```

**preflight check tool** 翻译成**自动检测工具**，有几个问题，1. 没有翻译出preflight，2. 原文没有自动的意思（虽然这个工具可能有），建议翻译为**预检工具**或者**预检测工具**。下同。

<!--
Note: The preflight tool only works for the x64 architecture. Fuchsia
is currently not guaranteed to build successfully on other host
architectures, such as Windows and ARM64.
-->

> 注意：这个检测工具只支持 x64 架构。Fuchsia 目前对其他架构的机器上的编译不做成功担保，比如 Windows 和 ARM64。

Reviewer: ```suggestion
注意：这个预检测工具只支持 x64 架构。目前 Fuchsia 不保证在其他宿主架构上能构建成功，比如 Windows 和 ARM64。
```
Fuchsia自己本身支持arm64，x64等架构，但是不支持在arm64的机器上构建，也就是他们开发环境的架构必须是x64，所以这里说的host是宿主的意思。

另外最后这句话原文作者应该写得有问题，Windows也支持x64架构，如果要包含Windows，应该说是不支持其他的平台。

<!--
Run the following command:
-->

> 运行下面的命令：

Reviewer: ```suggestion
运行如下命令：
```

<!--
# Download the Fuchsia source code
-->

> # 下载 Fuchsia 源码

Reviewer 2: ```suggestion
# 下载 Fuchsia 源代码
```

下文中“源码”请批量查找替换为“源代码”。

<!--
This guide provides instructions on how to download the
Fuchsia source code and set up the Fuchsia development
environment on your machine.
-->
本指南说明了如何下载 Fuchsia 源码以及在您的机器上部署开发环境。
<!--
The steps are:
-->
步骤如下：
<!--
1. [Perform a preflight check](#perform-a-preflight-check).
2. [Install prerequisite packages](#install-prerequisite-packages).
3. [Download the Fuchsia source code](#download-the-fuchsia-source-code).
4. [Set up environment variables](#set-up-environment-variables).
5. [Configure firewall rules (Optional)](#configure-firewall-rules).
-->
1. [执行预检查](#perform-a-preflight-check).
2. [安装必备软件包](#install-prerequisite-packages).
3. [下载 Fuchsia 源码](#download-the-fuchsia-source-code).
4. [设置环境变量](#set-up-environment-variables).
5. [配置防火墙规则（可选）](#configure-firewall-rules).

<!--
## 1. Perform a preflight check {#perform-a-preflight-check}
-->
## 1. 执行预检查 {#perform-a-preflight-check}
<!--
Fuchsia provides a preflight check tool
([`ffx platform preflight`][ffx-platform-preflight])
that examines your machine and informs you of any issues that may
affect building Fuchsia from source on the machine.
-->

> Fuchsia 提供了一个预检工具（[`ffx platform preflight`][ffx-platform-preflight]），这个工具可以测试您的机器，并会通知您任何在该机器上构建 Fuchsia 源码时会受影响的问题。

Reviewer 2: ```suggestion
Fuchsia 提供了一个预检查工具（[`ffx platform preflight`][ffx-platform-preflight]），这个工具可以测试您的机器，并会通知您在该机器上可能会影响从源代码构建 Fuchsia 的任何问题。
```

<!--
Note: The preflight tool only works for the x64 architecture. Fuchsia
is currently not guaranteed to build successfully on other host
architectures, such as Windows and ARM64.
-->

> 注意：这个预检测工具只支持 x64 架构。目前 Fuchsia 不保证在其他宿主架构上能构建成功，比如 Windows 和 ARM64。

Reviewer 2: ```suggestion
注意：这个预检查工具只支持 x64 架构。目前 Fuchsia 不保证在其他宿主架构上能构建成功，比如 Windows 和 ARM64。
```

前后没有一致。

  ```posix-terminal
  curl -sO https://storage.googleapis.com/fuchsia-ffx/ffx-macos-x64 && chmod +x ffx-macos-x64 && ./ffx-macos-x64 platform preflight
  ```
<!--
## 2. Install prerequisite packages {#install-prerequisite-packages}
-->

> ## 2. 安装用到的软件包 {#install-prerequisite-packages}

Reviewer: ```suggestion
## 2. 安装必备软件包 {#install-prerequisite-packages}
```

<!--
Fuchsia requires `curl`, `file`, `unzip`, and `git` to be up to date. The version
of `git` needs to be 2.28 or higher.
-->
Fuchsia 要求 `curl`、 `file`、 `unzip` 以及 `git` 等工具是最新版。 `git` 版本要求 2.28 或以上。

* {Linux}
<!--
  Install (or update) the following packages:
-->

>   安装（或更新）下面的软件：

Reviewer: ```suggestion
  安装（或更新）下面的软件包：
```

Ubuntu等Debian系Linux发行版的中文版package一般翻译成软件包，http://people.ubuntu.com/~happyaron/udc-cn/lucid-html/ch06s07.html 其中的**新立得软件包管理器**是https://en.wikipedia.org/wiki/Synaptic_(software)

下文所述的apt命令正是Debian系Linux发行版的特征之一。

  ```posix-terminal
  sudo apt install curl file git unzip
  ```

* {macOS}
<!--
  Install the Xcode command line tools:

  Note: Skip this step if `ffx platform preflight` shows that Xcode tools
  are already installed on your machine.
-->
  安装 Xcode 命令行工具：

>   注意：如果 `ffx platform preflight` 工具检测到 Xcode 已经安装，就省去这一步操作。

Reviewer: ```suggestion
  注意：如果 `ffx platform preflight` 检测到 Xcode 工具已经安装，就可以跳过这一步。
```

  ```posix-terminal
  xcode-select --install
  ```
<!--
## 3. Download the Fuchsia source code {#download-the-fuchsia-source-code}
-->
## 3. 下载 Fuchsia 源码 {#download-the-fuchsia-source-code}
<!--
Fuchsia provides a [bootstrap script](/scripts/bootstrap) that creates a
directory named `fuchsia` and downloads the Fuchsia source code in that
directory.
-->
Fuchsia 提供了一个[引导脚本](/scripts/bootstrap)，这个脚本会创建一个 `fuchsia` 的文件夹并把 Fuchsia 源码下载到这里。
<!--
Downloading the Fuchsia source code requires about 2 GB of space
on your machine. Depending on your build configuration, you need
another 80 to 90 GB of space later when you build Fuchsia. Additionally,
the download process uses a substantial amount of memory. It is advisible
to close non-crucial processes during this time.
-->

> 下载 Fuchsia 源码要求您机器上有 2 GB 的存储空间。之后在编译 Fuchsia 过程中需要另外 80 到 90 GB 的存储空间，这取决于您的编译选项配置。另外，下载过程中也会使用一部分的内存空间。建议您在此过程中关闭其他非必要的程序进程。

Reviewer: ```suggestion
下载 Fuchsia 源码要求您机器上有 2 GB 的存储空间。取决于您的构建配置，之后的构建 Fuchsia 过程则需要额外 80 到 90 GB 的存储空间。另外，下载过程中也会使用大量的内存。建议您在此过程中关闭其他非必要的程序进程。
```
build并不只是编译，编译只是build过程中的一种步骤，所以使用构建翻译build。
substantial amount of memory是**大量的内存**，建议多使用翻译工具核对单词含义。

<!--
To download the Fuchsia source, do the following:
-->
下载 Fuchsia 源码，请按如下步骤：
<!--
1.  Select a directory for downloading the Fuchsia source code, for example:

    Note: You can set up Fuchsia in any directory. This guide selects
    the `$HOME` directory as an example.
-->

> 1.  选择一个 Fuchsia 源码存储目录，例如：

Reviewer: ```suggestion
1.  选择一个 Fuchsia 源码下载目录，例如：
```

    
    注意：您可以把 Fuchsia 下载到任何路径。本篇教程选择 `$HOME` 作为示例。

    ```posix-terminal
    cd $HOME
    ```
<!--
1.  Run the bootstrap script:

    Note: Downloading the Fuchsia source code can take up to 60 minutes.
-->
1.  运行引导脚本：
    
    注意：下载 Fuchsia 源码可能需要 60 分钟。

    ```posix-terminal
    curl -s "https://fuchsia.googlesource.com/fuchsia/+/HEAD/scripts/bootstrap?format=TEXT" | base64 --decode | bash
    ```
<!--
    This script creates the `fuchsia` directory and downloads the source code.

    If you see the `Invalid authentication credentials` error during the
    bootstrapping process, see [Authentication error](#authentication-error) for
    help.
-->

>     这个脚本会先创建 `fuchsia` 路径，然后下载源码到该路径下。

Reviewer: ```suggestion
    这个脚本会创建 `fuchsia` 目录，并且下载源码。
```

注意前后统一，前面directory用了目录，这里又使用了路径，当然这里并没有路径的意思，一般path才翻译成路径。

>     如果您在脚本运行期间看到了 `Invalid authentication credentials` 错误信息，参考[认证失败](#authentication-error)章节寻求帮助。

Reviewer: ```suggestion
    如果您在脚本运行期间看到了 `Invalid authentication credentials` 错误信息，参考[认证错误](#authentication-error)章节寻求帮助。
```

  ```posix-terminal
  curl -sO https://storage.googleapis.com/fuchsia-ffx/ffx-macos-x64 && chmod +x ffx-macos-x64 && ./ffx-macos-x64 platform preflight
  ```

> <!--

Reviewer 2: ```suggestion

<!--
```
少了空行。会影响一些 md 可视化的渲染。

## 2. Install prerequisite packages {#install-prerequisite-packages}
-->
## 2. 安装必备软件包 {#install-prerequisite-packages}
<!--
Fuchsia requires `curl`, `file`, `unzip`, and `git` to be up to date. The version
of `git` needs to be 2.28 or higher.
-->

> Fuchsia 要求 `curl`、 `file`、 `unzip` 以及 `git` 等工具是最新版。 `git` 版本要求 2.28 或以上。

Reviewer 2: ```suggestion
Fuchsia 要求 `curl`、 `file`、 `unzip` 以及 `git` 等工具是最新版。 `git` 版本需为 2.28 或以上。
```

* {Linux}

> <!--

Reviewer 2: ```suggestion

<!--
```

不要占用掉原有的空行。

  Install (or update) the following packages:
-->

>   安装（或更新）下面的软件包：

Reviewer 2: ```suggestion
  安装（或更新）以下软件包：
```

  ```posix-terminal
  sudo apt install curl file git unzip
  ```

* {macOS}

> <!--

Reviewer 2: ```suggestion

<!--
```

  Install the Xcode command line tools:

  Note: Skip this step if `ffx platform preflight` shows that Xcode tools
  are already installed on your machine.
-->
  安装 Xcode 命令行工具：

>   注意：如果 `ffx platform preflight` 检测到 Xcode 工具已经安装，就可以跳过这一步。

Reviewer 2: ```suggestion
  注意：如果 `ffx platform preflight` 显示 Xcode 工具已经安装，则请跳过这一步。
```

翻译说法不太对。

  ```posix-terminal
  xcode-select --install
  ```

> <!--

Reviewer 2: ```suggestion

<!--
```

## 3. Download the Fuchsia source code {#download-the-fuchsia-source-code}
-->
## 3. 下载 Fuchsia 源码 {#download-the-fuchsia-source-code}
<!--
Fuchsia provides a [bootstrap script](/scripts/bootstrap) that creates a
directory named `fuchsia` and downloads the Fuchsia source code in that
directory.
-->

> Fuchsia 提供了一个[引导脚本](/scripts/bootstrap)，这个脚本会创建一个 `fuchsia` 的文件夹并把 Fuchsia 源码下载到这里。

Reviewer 2: ```suggestion
Fuchsia 提供了一个[引导脚本](/scripts/bootstrap)，这个脚本会创建一个名为 `fuchsia` 的文件夹，并把 Fuchsia 源码下载到这里。
```

<!--
Downloading the Fuchsia source code requires about 2 GB of space
on your machine. Depending on your build configuration, you need
another 80 to 90 GB of space later when you build Fuchsia. Additionally,
the download process uses a substantial amount of memory. It is advisible
to close non-crucial processes during this time.
-->

> 下载 Fuchsia 源码要求您机器上有 2 GB 的存储空间。取决于您的构建配置，之后的构建 Fuchsia 过程则需要额外 80 到 90 GB 的存储空间。另外，下载过程中也会使用大量的内存。建议您在此过程中关闭其他非必要的程序进程。

Reviewer 2: ```suggestion
下载 Fuchsia 源码要求您机器上有大约 2 GB 的存储空间。根据您的构建配置，之后在构建 Fuchsia 时，您还额外需要 80 到 90 GB 的存储空间。另外，下载过程中也会使用大量的内存。建议您在此过程中关闭其他非必要的进程。
```

<!--
To download the Fuchsia source, do the following:
-->

> 下载 Fuchsia 源码，请按如下步骤：

Reviewer 2: ```suggestion
要下载 Fuchsia 源代码，请执行如下步骤：
```

<!--
1.  Select a directory for downloading the Fuchsia source code, for example:

    Note: You can set up Fuchsia in any directory. This guide selects
    the `$HOME` directory as an example.
-->
1. 选择一个 Fuchsia 源码下载目录，例如：
    

>     注意：您可以把 Fuchsia 下载到任何路径。本篇教程选择 `$HOME` 作为示例。

Reviewer 2: ```suggestion
    注意：您可以在任何目录下设置 Fuchsia。本指南选择 `$HOME` 作为示例。
```

set up 是在说后续过程，即方便后续配置，不单纯指下载。

directory 目录；path 路径。

guide 指南；tutorial 教程。

还请注意。

    ```posix-terminal
    cd $HOME
    ```

> <!--

Reviewer 2: ```suggestion

<!--
```

1.  Run the bootstrap script:

    Note: Downloading the Fuchsia source code can take up to 60 minutes.
-->
1.  运行引导脚本：
    
    注意：下载 Fuchsia 源码可能需要 60 分钟。

    ```posix-terminal
    curl -s "https://fuchsia.googlesource.com/fuchsia/+/HEAD/scripts/bootstrap?format=TEXT" | base64 --decode | bash
    ```
<!--
    This script creates the `fuchsia` directory and downloads the source code.

    If you see the `Invalid authentication credentials` error during the
    bootstrapping process, see [Authentication error](#authentication-error) for
    help.
-->
    这个脚本会创建 `fuchsia` 目录，并且下载源码。

>     如果您在脚本运行期间看到了 `Invalid authentication credentials` 错误信息，参考[认证错误](#authentication-error)章节寻求帮助。

Reviewer 2: ```suggestion
    如果您在脚本运行期间看到了 `Invalid authentication credentials` 错误信息，请参考[认证错误](#authentication-error)章节寻求帮助。
```

<!--
## 4. Set up environment variables {#set-up-environment-variables}
-->
## 4. 设置环境变量 {#set-up-environment-variables}
<!--
Fuchsia recommends that you update your shell profile to include the following
actions:
-->

> Fuchsia 建议您在 shell 的配置文件中添加如下几个配置：

Reviewer 2: ```suggestion
Fuchsia 建议您按照如下操作更新 shell 配置文件：
```

可以参考，以减少和原文的差异。

    the Fuchsia project, and the `fx` tool helps configure, build, run, and
    debug Fuchsia. The Fuchsia toolchain requires that `jiri` is available in
    your `PATH`.
-->

> *  添加 `.jiri_root/bin` 路径到您的 `PATH` 环境变量

Reviewer: ```suggestion
*  添加 `.jiri_root/bin` 目录到您的 `PATH` 环境变量
```

>    Fuchsia 源码中的 `.jiri_root/bin` 目录包含了 [`jiri`](https://fuchsia.googlesource.com/jiri) 和 [`fx`](/development/build/fx.md) 工具，这些是 Fuchsia 工作流中的必备工具。Fuchsia 使用 `jiri` 工具在 Fuchsia 的项目中管理仓库，使用 `fx` 工具帮助配置、构建、运行以及调试 Fuchsia。Fuchsia 的工具链需要可以在您的 `PATH` 环境变量中找到 `jiri` 工具。

Reviewer 2: ```suggestion
   Fuchsia 源码中的 `.jiri_root/bin` 目录包含了 [`jiri`](https://fuchsia.googlesource.com/jiri) 和 [`fx`](/development/build/fx.md) 工具，这些是 Fuchsia 工作流中的必备工具。Fuchsia 使用 `jiri` 工具在 Fuchsia 的项目中管理仓库，而 `fx` 工具能够帮助配置、构建、运行以及调试 Fuchsia。Fuchsia 的工具链需要可以在您的 `PATH` 环境变量中找到 `jiri` 工具。
```

fx 那一句主语是 fx 不是 Fuchsia。不然就成了“Fuchsia 运行 Fuchsia”这样的描述了。

>    `.jiri_root/bin` 路径在 Fuchsia 源码中包含了 [`jiri`](https://fuchsia.googlesource.com/jiri) 和 [`fx`](/development/build/fx.md) 工具，这些工具对 Fuchsia 的后续操作起着至关重要的作用。Fuchsia 使用 `jiri` 工具在 Fuchsia 的项目中管理仓库，使用 `fx` 工具帮助配置、编译、运行以及调试 Fuchsia。Fuchsia 的工具链需要可以在您 `PATH` 环境变量中找得到 `jiri` 工具。

Reviewer: ```suggestion
  Fuchsia 源码中的 `.jiri_root/bin` 目录包含了 [`jiri`](https://fuchsia.googlesource.com/jiri) 和 [`fx`](/development/build/fx.md) 工具，这些是 Fuchsia 工作流中的必备工具。Fuchsia 使用 `jiri` 工具在 Fuchsia 的项目中管理仓库，使用 `fx` 工具帮助配置、构建、运行以及调试 Fuchsia。Fuchsia 的工具链需要可以在您的 `PATH` 环境变量中找到 `jiri` 工具。
```

    `FUCHSIA_DIR` environment variable and provides the `fd` command for
    navigating directories with auto-completion. (For more information, see
    comments in `fx-env.sh`.)
-->

> *   导入 `scripts/fx-env.sh` 文件

Reviewer: ```suggestion
*   导入 `scripts/fx-env.sh` 文件
```

这个source翻译成导入有点tricky，英文原文的话我们可以很容易猜测这是要在命令行里跑一下 `source scripts/fx-env.sh` ，但是把source翻译了之后中文环境就看不出这个猜测了。建议保留原文。参见fx-env.sh的注释https://cs.opensource.google/fuchsia/fuchsia/+/main:scripts/fx-env.sh;l=17?q=fx-env.sh&ss=fuchsia

Author: 个人建议翻译的，因为这是指一个动作，而不是专有名词。如果不翻译就属于是翻译人员的懒惰了。

Reviewer: 感觉可以添加一个译注。@y2-Whyto

Reviewer 2: ```suggestion
*   使用“source”命令导入 `scripts/fx-env.sh` 文件
```

这样翻译如何？直接括号标注的话可能也会让人猛一下有些摸不到头脑，因为是命令的名称，原文有些“活用”了，所以不如在中文里直接补充完整。

>     虽然这并不是必须的，导入 [`fx-env.sh`](/scripts/fx-env.sh) 文件可以在您的终端中建立一系列的非常有用的 shell 函数。比如，它会创建给 `fd` 工具用来切换路径时提供自动补全的 `FUCHSIA_DIR` 环境变量（如果想获取更多信息，请参考 `fx-env.sh` 文件）。

Reviewer: ```suggestion
    虽然这并不是必须的，导入 [`fx-env.sh`](/scripts/fx-env.sh) 文件可以在您的终端中启用一系列有用的 shell 函数。比如，它会创建 `FUCHSIA_DIR` 环境变量，以及提供 `fd` 命令用来在目录中导航时提供自动补全 （如果想获取更多信息，请参考 `fx-env.sh` 文件）。
```

创建FUCHSIA_DIR和fd是两个独立的事情，虽然确实fd依赖FUCHSIA_DIR，但是FUCHSIA_DIR在其他很多地方都会被用到，并不只是用于fd。

>     虽然这并不是必须的，导入 [`fx-env.sh`](/scripts/fx-env.sh) 文件可以在您的终端中启用一系列有用的 shell 函数。比如，它会创建 `FUCHSIA_DIR` 环境变量，以及提供 `fd` 命令用来在目录中导航时提供自动补全 （如果想获取更多信息，请参考 `fx-env.sh` 文件）。

Reviewer 2: ```suggestion
    虽然这并不是必须的，但是使用“source”命令导入 [`fx-env.sh`](/scripts/fx-env.sh) 文件可以在您的终端中启用一系列有用的 shell 函数。比如，它会创建 `FUCHSIA_DIR` 环境变量，以及提供 `fd` 命令用来在目录中导航时提供自动补全 （要获取更多信息，请参阅 `fx-env.sh` 中的注释）。
```

Author: 感觉复杂了，但可以接受

Reviewer 2: 189 行更新了更完整的修改

<!--
Note: If you don't wish to update your shell profile, see
[Work on Fuchsia without updating your PATH](#work-on-fuchsia-without-updating-your-path)
in Appendices instead.
-->

> 注意：如果您不想更新您的 shell 配置，在附录中参考[不更新 PATH 变量编译 Fuchsia](#work-on-fuchsia-without-updating-your-path)。

Reviewer: ```suggestion
注意：如果您不想更新您的 shell 配置，参考附录[不更新 PATH 变量的情况下在 Fuchsia 上工作](#work-on-fuchsia-without-updating-your-path)。
```

work on Fuchsia不能直接翻译成编译，拿到source也不一定会编译，比如我们做一些翻译工作，也算是work on fuchsia了，虽然是非官方的。

Author: 这样翻译的话读者容易造成误解

Reviewer: 请说明容易造成误解的理由。

Author: 可能会误解工作为使用 Fuchsia 系统中做别的工作上的事情。

Reviewer: 我觉得就算英文原句也有这个歧义。@y2-Whyto

> 注意：如果您不想更新您的 shell 配置，参考附录[不更新 PATH 变量的情况下在 Fuchsia 上工作](#work-on-fuchsia-without-updating-your-path)。

Reviewer 2: ```suggestion
注意：如果您不想更新您的 shell 配置，则请参阅[在不更新 PATH 变量的情况下处理 Fuchsia](#work-on-fuchsia-without-updating-your-path)。
```

“处理”有待讨论。

<!--
To update your shell profile to configure Fuchsia's environment variables,
do the following:
-->

> 为了更新您的 shell 配置文件来设置 Fuchsia 的环境变量，请按如下步骤：

Reviewer 2: ```suggestion
要更新您的 shell 配置文件来设置 Fuchsia 的环境变量，请执行如下步骤：
```

<!--
1.  Use a text editor to open your `~/.bash_profile` file (in the example below,
    we use the [Nano][nano]{:.external} text editor):

    Note: This guide uses a `bash` terminal as an example. If you're
    using `zsh`, replace `~/.bash_profile` with `~/.zprofile` in the
    following steps:
-->

> 1.  请用文本编辑器打开您的 `~/.bash_profile` 文件（参考如下示例，我们使用 [Nano][nano]{:.external} ）

Reviewer: ```suggestion
1.  请用文本编辑器打开您的 `~/.bash_profile` 文件（参考如下示例，我们使用 [Nano][nano]{:.external} 文本编辑器）
```

> 1.  请用文本编辑器打开您的 `~/.bash_profile` 文件（参考如下示例，我们使用 [Nano][nano]{:.external} 文本编辑器）

Reviewer 2: ```suggestion
1.  使用文本编辑器打开您的 `~/.bash_profile` 文件（在以下示例中，我们使用 [Nano][nano]{:.external} 文本编辑器）：
```

列表里这种行为描述，前后一致起见就先都不用“请”了。

漏标点。

>     注意：这篇文章使用 `bash` 终端作为示例，如果您使用 `zsh` 的话，在如下步骤中请替换文件 `~/.bash_profile` 为 `~/.zprofile`：

Reviewer: ```suggestion
    注意：本指南使用 `bash` 终端作为示例，如果您使用 `zsh` 的话，在如下步骤中请替换 `~/.bash_profile` 为 `~/.zprofile`：
```

>     注意：本指南使用 `bash` 终端作为示例，如果您使用 `zsh` 的话，在如下步骤中请替换 `~/.bash_profile` 为 `~/.zprofile`：

Reviewer 2: ```suggestion
    注意：本指南使用 `bash` 终端作为示例，如果您使用 `zsh`，请在后续步骤中请替换 `~/.bash_profile` 为 `~/.zprofile`：
```

不要出现“的话”这种口语词。

“后续”可能比“如下”能管到的范围更远一些。

    ```posix-terminal
    nano ~/.bash_profile
    ```

> <!--

Reviewer 2: ```suggestion

<!--
```

1.  Add the following lines to your `~/.bash_profile` file:

    Note: If your Fuchsia source code is not located in the `~/fuchsia`
    directory, replace `~/fuchsia` with your Fuchsia directory.
-->
1.  在您的 `~/.bash_profile` 文件中添加如下配置：

>     注意：如果您的 Fuchsia 源码不在 `~/fuchsia` 路径下，替换 `~/fuchsia` 路径为您的 Fuchsia 路径。

Reviewer: ```suggestion
    注意：如果您的 Fuchsia 源码不在 `~/fuchsia` 目录下，替换 `~/fuchsia` 为您的 Fuchsia 目录。
```

>     注意：如果您的 Fuchsia 源码不在 `~/fuchsia` 目录下，替换 `~/fuchsia` 为您的 Fuchsia 目录。

Reviewer 2: ```suggestion
    注意：如果您的 Fuchsia 源码不在 `~/fuchsia` 目录下，则请替换 `~/fuchsia` 为您的 Fuchsia 目录。
```

    ```sh
    export PATH=~/fuchsia/.jiri_root/bin:$PATH
    source ~/fuchsia/scripts/fx-env.sh
    ```

> <!--

Reviewer 2: ```suggestion

<!--
```

1.  Save the file and exit the text editor.
-->
1.  保存文件并退出。
<!--
1.  To update your environment variables, run the following command:
-->

> 1.  更新环境变量，请运行如下命令：

Reviewer 2: ```suggestion
1.  要更新环境变量，请运行如下命令：
```

“To”开头要加一个“要”或者“要想”。

    ```posix-terminal
    source ~/.bash_profile
    ```

> <!--

Reviewer 2: ```suggestion

<!--
```

1.  Verify that you can run the following commands inside your
    `fuchsia` directory without error:
-->

> 1.  验证您是否可以运行如下命令且没有报错：

Reviewer 2: ```suggestion
1.  验证您可以在您的 `fuchsia` 目录内运行如下命令且没有报错：
```

肯定。

漏译。

    ```posix-terminal
    fx help
    ```
<!--
## 5. Configure firewall rules (Optional) {#configure-firewall-rules}
-->

> ## 5. 配置防火墙（可选项） {#configure-firewall-rules}

Reviewer: ```suggestion
## 5. 配置防火墙规则（可选） {#configure-firewall-rules}
```

<!--
Note: This step is not required for building or running Fuchsia. But it is
recommended to ensure that Fuchsia's emulator instances run smoothly on Linux.

(**Linux only**) If you're planning on running Fuchsia on Linux, it is advised to
run the following command to allow Fuchsia-specific traffic on the host machine:
-->

> 注意：这一步在您的编译或者运行 Fuchsia 步骤中并不是必须的。但是是被建议的，因为可以用来确保您能在 Linux 中流畅的运行 Fuchsia 虚拟机。

Reviewer: ```suggestion
注意：这一步对构建或者运行 Fuchsia 并不是必须的。但是为了确保 Fuchsia 模拟器实例能在 Linux 上顺利运行是推荐的。
```

虚拟机是virtual machine，emulator是模拟器，建议多使用翻译工具确认单词意思。

Author: 意思都是一样的，指代同一种东西。不过如果有要求的话，就按照标准来吧。

> (**Linux**) 如果您计划在 Linux 中运行 Fuchsia，建议您运行如下命令来允许 Fuchsia 使用一些特殊的特性：

Reviewer: ```suggestion
(**仅Linux**) 如果您计划在 Linux 中运行 Fuchsia，建议您运行如下命令在宿主机上允许 Fuchsia 相关的流量：
```

traffic是流量或者交通，建议使用翻译工具或者字典确认含义。

```posix-terminal
fx setup-ufw
```
<!--
This script requires `sudo` (which asks for your password) to set the appropriate
firewall rules. (For more information on this script, see [`setup-ufw`][setup-ufw].)
-->
这个脚本需要 `sudo` （会提示您输入密码）权限来设置防火墙规则。（更多关于脚本的的信息，参考 [`setup-ufw`][setup-ufw] 文件）。
<!--
## Next steps
-->
## 下一步

<!--
To build your first Fuchsia system image, see
[Configure and build Fuchsia](/get-started/build_fuchsia.md).
-->

> 编译您第一个 Fuchsia 系统镜像，参考[配置和编译 Fuchsia](/get-started/build_fuchsia.md) 文件。

Reviewer: ```suggestion
构建您第一个 Fuchsia 系统镜像，参考[配置和构建 Fuchsia](/get-started/build_fuchsia.md)。
```

<!--
## Appendices
-->
## 附录
<!--
### Authentication error {#authentication-error}
-->

> ### 验证失败 {#authentication-error}

Reviewer: ```suggestion
### 认证错误 {#authentication-error}
```

注意前后保持一致，前文中已经使用**认证**来翻译authentication，这里却又换了一个说法。

<!--
If you see the `Invalid authentication credentials` error during the bootstrap
process, your `~/.gitcookies` file may contain cookies from some repositories in
`googlesource.com` that the bootstrap script wants to check out anonymously.
-->

> 如果您在脚本运行期间看到了 `Invalid authentication credentials` 报错信息，您的 `~/.gitcookies` 文件中可能包含了一些引导脚本想要自动检测的来自 `googlesource.com` 的 cookie 信息。

Reviewer: ```suggestion
如果您在引导脚本运行过程中看到了 `Invalid authentication credentials` 报错信息，您的 `~/.gitcookies` 文件中可能包含了来自 `googlesource.com` 中的一些仓库的 cookie，引导脚本想要匿名检出。
```

check out  -- 结合上下文这里谈的是bootstrap script，该脚本的作用是下载source code，因为fuchsia用的git，所以这里的check out应该是git的checkout，所以翻译为检出。https://linux.cn/article-12245-1.html

Author: 下载代码的话不一定是使用git工具的，同时 check out 一般指检查某一项指标或者任务是否达标或完成，尤其在脚本中。

Reviewer: [bootstrap脚本](https://cs.opensource.google/fuchsia/fuchsia/+/main:scripts/bootstrap?q=scripts%2Fbootstrap&ss=fuchsia) 是用[jiri](https://fuchsia.googlesource.com/jiri)管理一系列git repo和cipd，该脚本读取[flower](https://fuchsia.googlesource.com/integration/+/refs/heads/main/flower) 清单，最后下载的是[experience](https://fuchsia.googlesource.com/experiences/) git repo，结合这里的.gitcookies，这里就是说的git的check out。

Author: ok

<!--
To resolve this error, do one of the following:
-->
为了解决这个问题，使用以下任何一个方式：

<!--
*   Follow the onscreen directions to get passwords for the specified
    repositories.
*   Delete the offending cookies from the `.gitcookies` file.
-->

> *  按照屏幕路径的提示为仓库输入密码

Reviewer: ```suggestion
*  按照屏幕上的指示为特定的仓库获取密码
```

> *  删除 `.gitcookies` 文件中不需要的 cookie。

Reviewer: ```suggestion
*  删除 `.gitcookies` 文件中有问题的 cookie。
```

<!--
### Work on Fuchsia without updating your PATH {#work-on-fuchsia-without-updating-your-path}
-->

> ###  不更新 PATH 变量编译 Fuchsia {#work-on-fuchsia-without-updating-your-path}

Reviewer: ```suggestion
###  不更新 PATH 变量的情况下在 Fuchsia 上工作{#work-on-fuchsia-without-updating-your-path}
```

Author: 同上

Reviewer 2: 这里 work on 翻译成“在……工作”可能不太好，需要根据前后文翻译。就我查到的文档而言，通用的翻译是“使用”或“处理”，不过分别对应的宾语是某个软件和文档/文件。这里可以考虑“处理”？不知你们怎么感觉。

<!--
The following sections provide alternative approaches to the
[Set up environment variables](#set-up-environment-variables) section:
-->
下面的章节为[设置环境变量](#set-up-environment-variables)章节提供了替代选项：
<!--
*   [Copy the tool to your binary directory](#copy-the-tool-to-your-binary-directory)
*   [Add a symlink to your binary directory](#add-a-symlink-to-your-binary-directory)
-->

> *   [把工具复制到可执行文件目录](#copy-the-tool-to-your-binary-directory)

Reviewer: ```suggestion
*   [把工具复制到二进制目录](#copy-the-tool-to-your-binary-directory)
```

> *   [为工具添加链接文件到可执行文件目录](#add-a-symlink-to-your-binary-directory)

Reviewer: ```suggestion
*   [添加符号链接到二进制目录](#add-a-symlink-to-your-binary-directory)
```

<!--
#### Copy the tool to your binary directory {#copy-the-tool-to-your-binary-directory}
-->

> ####  把工具复制到可执行文件目录 {#copy-the-tool-to-your-binary-directory}

Reviewer: 同上。

<!--
If you don't wish to update your environment variables, but you want `jiri` to
work in any directory, copy the `jiri` tool to your `~/bin` directory, for
example:

Note: If your Fuchsia source code is not located in the `~/fuchsia` directory,
replace `~/fuchsia` with your Fuchsia directory.
-->
如果您不想更新您的环境变量，但是您想在任何目录中使用 `jiri` 工具，把 `jiri` 工具复制到 `~/bin` 目录中，比如：

> 注意：如果您的 Fuchsia 源码不在 `~/fuchsia` 目录中，把 `~/fuchsia` 替换为您的 Fuchsia 目录。

Reviewer: 注意前后一致，前文的翻译跟这里不一样，前文的修改建议跟这里的翻译是一样的。。。

```posix-terminal
cp ~/fuchsia/.jiri_root/bin/jiri ~/bin
```

<!--
However, you must have write access to the `~/bin` directory without `sudo`. If
you don't, `jiri` cannot keep itself up-to-date.
-->

> 无论如何，您首先需要有不用 `sudo` 命令就可以访问 `~/bin` 路径的权限。如果没有，`jiri` 就不能使它自己保持在最新版本。

Reviewer: ```suggestion
但是，您需要在没有 `sudo` 的情况下对 `~/bin` 有写访问权限。如果没有，`jiri` 就不能保持在最新版本。
```

<!--
#### Add a symlink to your binary directory {#add-a-symlink-to-your-binary-directory}
-->

> #### 为工具添加链接文件到可执行文件目录 {#add-a-symlink-to-your-binary-directory}

Reviewer: 同上。

<!--
Similarly, if you want to use the `fx` tool without updating your environment
variables, provide the `fx` tool's symlink in your `~/bin` directory, for
example:

Note: If your Fuchsia source code is not located in the `~/fuchsia` directory,
replace `~/fuchsia` with your Fuchsia directory.
-->

> 同样的，如果您想不更新环境变量使用 `fx` 工具，在 `~/bin` 路径中提供您的 `fx` 工具的链接文件，比如：

Reviewer: ```suggestion
同样的，如果您想不更新环境变量而使用 `fx` 工具，在 `~/bin` 目录中提供 `fx` 工具的符号链接，比如：
```

前后一致，directory在这篇翻译中**目录**和**路径**一直在混用。

Reviewer: 这一条并没有resolve？

注意：如果您的 Fuchsia 源码不在 `~/fuchsia` 目录中，把 `~/fuchsia` 替换为您的 Fuchsia 目录。

```posix-terminal
ln -s ~/fuchsia/scripts/fx ~/bin
```
<!--
Alternatively, run the `fx` tool directly using its path, for example:
-->

> 相应地，您也可以使用 `fx` 工具的完整路径，比如：

Reviewer: ```suggestion
或者，直接用其路径运行 `fx` 工具，比如：
```

相应是accordingly，alternatively是或者。请多使用字典或者翻译工具确认单词意思。

```posix-terminal
./scripts/fx help
```
<!--
In either case, you need `jiri` in your `PATH`.
-->

> 另一方面，您需要在 `PATH` 环境变零中设置您的 `jiri` 工具。

Reviewer: ```suggestion
无论哪种情况，您都需要`jiri` 工具在您的 `PATH` 环境变量中。
```

    ```posix-terminal
    fx help
    ```
<!--
## 5. Configure firewall rules (Optional) {#configure-firewall-rules}
-->
## 5. 配置防火墙规则（可选） {#configure-firewall-rules}
<!--
Note: This step is not required for building or running Fuchsia. But it is
recommended to ensure that Fuchsia's emulator instances run smoothly on Linux.

(**Linux only**) If you're planning on running Fuchsia on Linux, it is advised to
run the following command to allow Fuchsia-specific traffic on the host machine:
-->

> 注意：这一步对构建或者运行 Fuchsia 并不是必须的。但是为了确保 Fuchsia 模拟器实例能在 Linux 上顺利运行是推荐的。

Reviewer 2: ```suggestion
注意：这一步对构建或者运行 Fuchsia 并不是必需的。但是为了确保 Fuchsia 模拟器实例能在 Linux 上顺利运行是推荐的。
```

required 必“需”的

> 注意：这一步对构建或者运行 Fuchsia 并不是必需的。但是为了确保 Fuchsia 模拟器实例能在 Linux 上顺利运行是推荐的。

Reviewer 2: ```suggestion
注意：这一步对构建或者运行 Fuchsia 并不是必需的。但是推荐您进行该步骤，以确保 Fuchsia 模拟器实例能在 Linux 上流畅运行。
```

原译文有语法问题。另外“smoothly”是流畅地/顺畅地；而顺利地可能一般会用“successfully”。

> (**仅Linux**) 如果您计划在 Linux 中运行 Fuchsia，建议您运行如下命令在宿主机上允许 Fuchsia 相关的流量：

Reviewer 2: ```suggestion
(**仅限 Linux**) 如果您计划在 Linux 中运行 Fuchsia，建议您运行如下命令在宿主机上允许 Fuchsia 相关的流量：
```

空格。

> (**仅限 Linux**) 如果您计划在 Linux 中运行 Fuchsia，建议您运行如下命令在宿主机上允许 Fuchsia 相关的流量：

Reviewer 2: ```suggestion
(**仅限 Linux**) 如果您计划在 Linux 中运行 Fuchsia，那么建议您运行如下命令，在宿主机上允许 Fuchsia 特定流量：
```

Fuchsia-specific 可以考虑翻译为“Fuchsia 专用的”或者“Fuchsia 特定的”。

```posix-terminal
fx setup-ufw
```
<!--
This script requires `sudo` (which asks for your password) to set the appropriate
firewall rules. (For more information on this script, see [`setup-ufw`][setup-ufw].)
-->

> 这个脚本需要 `sudo` （会提示您输入密码）权限来设置防火墙规则。（更多关于脚本的的信息，参考 [`setup-ufw`][setup-ufw] 文件）。

Reviewer 2: ```suggestion
该脚本需要 `sudo` 权限（会要求您输入密码）来设置适当的防火墙规则。（要获取关于该脚本的更多信息，请参考 [`setup-ufw`][setup-ufw]）。
```

有不准确的地方。固定句式。

<!--
## Next steps
-->

> ## 下一步

Reviewer 2: ```suggestion
## 后续步骤
```

根据翻译术语表内容进行修改。

<!--
To build your first Fuchsia system image, see
[Configure and build Fuchsia](/get-started/build_fuchsia.md).
-->

> 构建您第一个 Fuchsia 系统镜像，参考[配置和构建 Fuchsia](/get-started/build_fuchsia.md)。

Reviewer 2: ```suggestion
要构建您第一个 Fuchsia 系统镜像，请参考[配置和构建 Fuchsia](/get-started/build_fuchsia.md)。
```

> 要构建您第一个 Fuchsia 系统镜像，请参考[配置和构建 Fuchsia](/get-started/build_fuchsia.md)。

Reviewer 2: ```suggestion
要构建您第一个 Fuchsia 系统镜像，请参阅[配置和构建 Fuchsia](/get-started/build_fuchsia.md)。
```

<!--
## Appendices
-->
## 附录
<!--
### Authentication error {#authentication-error}
-->
### 认证错误 {#authentication-error}
<!--
If you see the `Invalid authentication credentials` error during the bootstrap
process, your `~/.gitcookies` file may contain cookies from some repositories in
`googlesource.com` that the bootstrap script wants to check out anonymously.
-->

> 如果您在引导脚本运行过程中看到了 `Invalid authentication credentials` 报错信息，您的 `~/.gitcookies` 文件中可能包含了来自 `googlesource.com` 中的一些仓库的 cookie，引导脚本想要匿名检出。

Reviewer 2: ```suggestion
如果您在引导脚本运行过程中看到了 `Invalid authentication credentials` 报错信息，那么您的 `~/.gitcookies` 文件中可能包含了来自 `googlesource.com` 中一些仓库的 cookie，引导脚本想要匿名检出。
```

> 如果您在引导脚本运行过程中看到了 `Invalid authentication credentials` 报错信息，那么您的 `~/.gitcookies` 文件中可能包含了来自 `googlesource.com` 中一些仓库的 cookie，引导脚本想要匿名检出。

Reviewer 2: ```suggestion
如果您在引导脚本运行过程中看到了 `Invalid authentication credentials`（无效的认证凭据）错误信息，那么您的 `~/.gitcookies` 文件中可能含有来自 `googlesource.com` 中一些仓库的 cookie，引导脚本想要匿名检查。
```

<!--
To resolve this error, do one of the following:
-->

> 为了解决这个问题，使用以下任何一个方式：

Reviewer 2: ```suggestion
要解决该错误，请使用以下方式之一：
```

<!--
*   Follow the onscreen directions to get passwords for the specified
    repositories.
*   Delete the offending cookies from the `.gitcookies` file.
-->

> *  按照屏幕上的指示为特定的仓库获取密码

Reviewer: ```suggestion
*  按照屏幕上的指示为特定的仓库获取密码。
```

> *  按照屏幕上的指示为特定的仓库获取密码。

Reviewer 2: ```suggestion
*  按照屏幕上的指示为指定仓库获取密码。
```

specified 指定的；specific 具体的/特定的。

*  删除 `.gitcookies` 文件中有问题的 cookie。

<!--
### Work on Fuchsia without updating your PATH {#work-on-fuchsia-without-updating-your-path}
-->
###  不更新 PATH 变量编译 Fuchsia {#work-on-fuchsia-without-updating-your-path}
<!--
The following sections provide alternative approaches to the
[Set up environment variables](#set-up-environment-variables) section:
-->

> 下面的章节为[设置环境变量](#set-up-environment-variables)章节提供了替代选项：

Reviewer 2: ```suggestion
下面的章节为[设置环境变量](#set-up-environment-variables)章节提供了替代方法：
```

<!--
*   [Copy the tool to your binary directory](#copy-the-tool-to-your-binary-directory)
*   [Add a symlink to your binary directory](#add-a-symlink-to-your-binary-directory)
-->
*   [把工具复制到二进制目录](#copy-the-tool-to-your-binary-directory)
*   [添加符号链接到二进制目录](#add-a-symlink-to-your-binary-directory)

<!--
#### Copy the tool to your binary directory {#copy-the-tool-to-your-binary-directory}
-->
####  把工具复制到二进制目录 {#copy-the-tool-to-your-binary-directory}
<!--
If you don't wish to update your environment variables, but you want `jiri` to
work in any directory, copy the `jiri` tool to your `~/bin` directory, for
example:

Note: If your Fuchsia source code is not located in the `~/fuchsia` directory,
replace `~/fuchsia` with your Fuchsia directory.
-->

> 如果您不想更新您的环境变量，但是您想在任何目录中使用 `jiri` 工具，把 `jiri` 工具复制到 `~/bin` 目录中，比如：

Reviewer 2: ```suggestion
如果您不想更新您的环境变量，但是想在任何目录中使用 `jiri` 工具，那么请把 `jiri` 工具复制到 `~/bin` 目录中，比如：
```

> 注意：如果您的 Fuchsia 源码不在 `~/fuchsia` 目录中，把 `~/fuchsia` 替换为您的 Fuchsia 目录。

Reviewer 2: ```suggestion
注意：如果您的 Fuchsia 源码不在 `~/fuchsia` 目录中，则请把 `~/fuchsia` 替换为您的 Fuchsia 目录。
```

```posix-terminal
cp ~/fuchsia/.jiri_root/bin/jiri ~/bin
```

<!--
However, you must have write access to the `~/bin` directory without `sudo`. If
you don't, `jiri` cannot keep itself up-to-date.
-->

> 但是，您需要在没有 `sudo` 的情况下对 `~/bin` 有写访问权限。如果没有，`jiri` 就不能保持在最新版本。

Reviewer 2: ```suggestion
但是，您必须在没有 `sudo` 的情况下对 `~/bin` 有写访问权限。否则，`jiri` 无法自动保持最新版本。
```

<!--
#### Add a symlink to your binary directory {#add-a-symlink-to-your-binary-directory}
-->
#### 添加符号链接到二进制目录 {#add-a-symlink-to-your-binary-directory}
<!--
Similarly, if you want to use the `fx` tool without updating your environment
variables, provide the `fx` tool's symlink in your `~/bin` directory, for
example:

Note: If your Fuchsia source code is not located in the `~/fuchsia` directory,
replace `~/fuchsia` with your Fuchsia directory.
-->

> 同样的，如果您想不更新环境变量使用 `fx` 工具，在 `~/bin` 路径中提供您的 `fx` 工具的链接文件，比如：

Reviewer 2: ```suggestion
同样地，如果您想在不更新环境变量的情况下使用 `fx` 工具，则请在 `~/bin` 路径中添加您 `fx` 工具的链接文件，比如：
```

土也地。保持句子通顺、书面化。

> 同样的，如果您想不更新环境变量而使用 `fx` 工具，在 `~/bin` 目录中提供 `fx` 工具的符号链接，比如：

Reviewer 2: ```suggestion
```

重复了。

> 注意：如果您的 Fuchsia 源码不在 `~/fuchsia` 目录中，把 `~/fuchsia` 替换为您的 Fuchsia 目录。

Reviewer 2: ```suggestion
注意：如果您的 Fuchsia 源码不在 `~/fuchsia` 目录中，则请把 `~/fuchsia` 替换为您的 Fuchsia 目录。
```

```posix-terminal
ln -s ~/fuchsia/scripts/fx ~/bin
```
<!--
Alternatively, run the `fx` tool directly using its path, for example:
-->

> 或者，直接用其路径运行 `fx` 工具，比如：

Reviewer 2: ```suggestion
或者，请直接使用 `fx` 工具的路径来运行，比如：
```

```posix-terminal
./scripts/fx help
```
<!--
In either case, you need `jiri` in your `PATH`.
-->

> 无论哪种情况，您都需要`jiri` 工具在您的 `PATH` 环境变量中。

Reviewer: ```suggestion
无论哪种情况，您都需要 `jiri` 工具在您的 `PATH` 环境变量中。
```
